Drivers/ProductAttributesPartDriver.cs

```diff
@@ -58,7 +58,11 @@ public class ProductAttributesPartDriver : ContentPartDriver<ProductAttributesPa
             // If the part isn't there, there must be no attributes
             if (attributesPart == null) return attributeIdsToValues == null || !attributeIdsToValues.Any();
             // If the part is there, it must have as many attributes as were passed in
-            if (attributesPart.AttributeIds.Count() != attributeIdsToValues.Count) return false;
+            if (attributesPart.AttributeIds.Count() != attributeIdsToValues.Count) {
+                //Attributes may have been deleted
+                attributesPart.AttributeIds = _attributeService.GetAttributes(attributeIdsToValues.Keys).Select(pap => pap.Id);

```

Comment:
Looks like I based this branch off the changes I made for #91. This comes from that.